src/components/application_manager/include/application_manager/application.h

```diff
@@ -786,6 +791,23 @@ class Application : public virtual InitialApplicationData,
    */
   virtual uint32_t GetAvailableDiskSpace() = 0;
 
+  /**
+   * @breif Allows to save mobile's command smart object in order to perform
+   * this command later.
+   * @param mobile_message the message smart_object.
+   */
+  virtual void PushMobileMessage(
+      smart_objects::SmartObjectSPtr mobile_message) = 0;
+
+  /**
+   * @breif Allows to obtain the whole list of pending commands in order to
+   * process them.
+   * @param mobile_message the messages array which is filled by the method.
+   * @Note After the method return the data internal messages array will be
```

Comment:
@LitvinenkoIra please modify description.
Internal message will be swapped with mobile_messages parameter, but not cleared